packages/squiggle-lang/src/rescript/FR/FR_Pointset.res

```diff
@@ -4,21 +4,28 @@ open FunctionRegistry_Helpers
 let nameSpace = "PointSet"
 let requiresNamespace = true
 
-let inputsTodist = (inputs: array<FunctionRegistry_Core.frValue>, makeDist) => {
-  let array = inputs->getOrError(0)->E.R.bind(Prepare.ToValueArray.Array.openA)
-  let xyCoords =
-    array->E.R.bind(xyCoords =>
-      xyCoords
-      ->E.A2.fmap(xyCoord =>
-        [xyCoord]->Prepare.ToValueArray.Record.twoArgs->E.R.bind(Prepare.ToValueTuple.twoNumbers)
-      )
-      ->E.A.R.firstErrorOrOpen
-    )
-  let expressionValue =
-    xyCoords
-    ->E.R.bind(r => r->XYShape.T.makeFromZipped->E.R2.errMap(XYShape.Error.toString))
-    ->E.R2.fmap(r => Reducer_T.IEvDistribution(PointSet(makeDist(r))))
-  expressionValue
+let inputsToDist = (inputs: array<Reducer_T.value>, xyShapeToPointSetDist) => {
+  // TODO - rewritein more functional/functor-based style
+  switch inputs {
+    | [IEvArray(items)] => {
+      items->Belt.Array.map(
+        item =>
+        switch item {
+          | IEvRecord(map) => {
+            let xValue = map->Belt.Map.String.getExn("x")
```

Comment:
if you have a switch right below anyway, I don't get the point of doing ``getExn``. I'd just do a ``get``.

Reply:
Thanks, fixed!